src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -275,8 +275,20 @@ protected function insert_attachment( $request ) {
 		$attachment->post_mime_type = $type;
 		$attachment->guid           = $url;
 
+		// If the title was not set, use the filename extracted from the request header.
 		if ( empty( $attachment->post_title ) ) {
-			$attachment->post_title = preg_replace( '/\.[^.]+$/', '', wp_basename( $file ) );
+			$content_disposition = $request->get_header( 'content_disposition' );
+
+			if ( ! empty( $content_disposition ) ) {
+				$lower_content_disposition = strtolower( $content_disposition );
+
+				if ( 0 === strpos( $lower_content_disposition, 'attachment; filename=' ) ) {
```

Comment:
```suggestion
				if ( str_starts_with( $lower_content_disposition, 'attachment; filename=' ) ) {
```
We're aiming to use `str_starts_with()` and the like from now on. [Reference](https://core.trac.wordpress.org/ticket/58012)

Alternatively, you could remove the call to `strtolower()` above and just use `0 === stripos()`, as `$lower_content_disposition` is only used for this check.